### ISSUE TITLE
Suggestion: Add overloads for AddEventListener methods

### ISSUE THREAD

**Issue by strygo** [bridgedotnet/Bridge#897]
Ideally, there would be overloads similar to:

```
AddEventListener<TEvent>(string name, Action<TEvent> callback) where TEvent : Event
```

**Comment by geoffreymcgill** [comment]
We discussed this issue today, although need to do a little research.

**Comment by DaniilVeriga** [comment]
@strygo, could you, please, clarify a use case? I don't say there is no use case. Certainly, there is. Also it might be obvious what a use case is, but it would be ideal if you clarify the exact scenario where it is needed.

**Comment by strygo** [comment]
```
Window.AddEventListener<ErrorEvent>("error", this.OnErrorEvent);

private void OnErrorEvent(ErrorEvent evt)
{

}
```

**Comment by DaniilVeriga** [comment]
So, currently, it is possible to do with

```
Window.AddEventListener("error", (Action<ErrorEvent>)this.OnErrorEvent);
```
but you'd like to avoid casting and prefer a generic approach, right?

By the way, I have noticed that there is no ErrorEvent in Bridge.Html5. Created an issue - #941.

**Comment by strygo** [comment]
yes, I'd like to avoid the casts. Another interesting idea might be to map:

Window.AddErrorEventListener(Action<ErrorEvent> callback);

This would generally require more overloads for various events but it would make them all more type safe.
